go/oasis-node/cmd/debug/byzantine/executor.go

```diff
@@ -116,24 +127,20 @@ func (cbc *computeBatchContext) proposeTransactionBatch(ctx context.Context, gro
 		StorageSignatures: ioReceiptSignatures,
 		Header:            currentBlock.Header,
 	}
+	// Corrupt the proposed batch.
+	if corrupt {
+		dispatchMsg.IORoot = emptyRoot.Hash
```

Comment:
So this corrupts it in a way that receipts are also invalidated and as such the signature verification will fail.

You could also corrupt it in other ways (at least until #3440 is implemented as then storage nodes would reject these):
* Storing garbage in the IO root while keeping the receipts valid. Nodes should already correctly reject such proposals.
* Putting more transactions in the IO root than allowed by `MaxBatchSize`/`MaxBatchSizeBytes`. Nodes should already correctly reject such proposals.

Reply:
Added a comment about how this corrupts the proposed batch.  Yeah probably doesn't make sense adding the mentioned corruptions since they'll soon be already rejected by the storage worker.

An interesting corruption would be something that passes all the checks but would only fail at runtime processing, which should results in all other nodes submitting failure indicating commitments -  but i don't think there's a way to do this (it'll always be caught before).